wetterdienst/dwd/observations/api.py

```diff
@@ -238,6 +294,103 @@ def collect_safe(self):
 
         return pd.concat(data)
 
+    def _invalidate_storage(self):
+        for parameter in self.parameter:
+            for period_type in self.period_type:
+                storage = self.storage.hdf5(
+                    parameter=parameter,
+                    time_resolution=self.time_resolution,
+                    period_type=period_type,
+                )
+
+                storage.invalidate()
+
+    @staticmethod
+    def _tidy_up_data(df: pd.DataFrame, parameter: Parameter) -> pd.DataFrame:
+        """
+        Function to create a tidy DataFrame by reshaping it, putting quality in a
+        separate column and setting an extra column with the parameter.
+
+        :param df:          DataFrame to be tidied
+        :param parameter:   the parameter that is written in a column to identify a set
+                            of different parameters amongst each other
+
+        :return:            The tidied DataFrame
+        """
+        id_vars = []
+        date_vars = []
+
+        # Add id columns based on metadata columns
+        for column in POSSIBLE_ID_VARS:
+            if column in df:
+                id_vars.append(column)
+                if column in POSSIBLE_DATE_VARS:
+                    date_vars.append(column)
+
+        # Extract quality
+        # Set empty quality for first columns until first QN column
+        quality = pd.Series()
+        column_quality = pd.Series()
+
+        for column in df:
+            # If is quality column, overwrite current "column quality"
+            if column.startswith("QN"):
+                column_quality = df.pop(column)
+            else:
+                quality = quality.append(column_quality)
+
+        df_tidy = df.melt(
+            id_vars=id_vars,
+            var_name=DWDMetaColumns.ELEMENT.value,
+            value_name=DWDMetaColumns.VALUE.value,
+        )
+
+        df_tidy[DWDMetaColumns.PARAMETER.value] = parameter.name
+
+        df_tidy[DWDMetaColumns.QUALITY.value] = quality.reset_index(drop=True).astype(
+            pd.Int64Dtype()
+        )
+
+        # Reorder properly
+        df_tidy = df_tidy.reindex(
+            columns=[
+                DWDMetaColumns.STATION_ID.value,
+                DWDMetaColumns.PARAMETER.value,
+                DWDMetaColumns.ELEMENT.value,
+                *date_vars,
+                DWDMetaColumns.VALUE.value,
+                DWDMetaColumns.QUALITY.value,
+            ]
+        )
+
+        return df_tidy
+
+    @staticmethod
+    def _create_humanized_column_names_mapping(
+        time_resolution: TimeResolution, parameter: Parameter
+    ) -> dict:
```

Comment:
Because this function a) is static and b) uses `DWDOrigDataColumns` and `DWDDataColumns`, i.e. it is just a transformer function to both, we might well just put it somewhere into `wetterdienst.dwd.metadata`.

Reply:
I had relocated it to the class as it is not used anywhere else currently. But can move it back again